host-spec/c04-networking.tm

```diff
@@ -81,551 +128,427 @@
+    data for the requested sequence>>|<row|<cell|BlockData>|<cell|>|<cell|>>>>>>
+      <verbatim|BlockResponse> Protobuf message.
+    </big-table>
+
+    where <verbatim|BlockData> is a Protobuf structure containing the
+    requested blocks. Do note that the optional values are either present or
+    absent depending on the requested fields (bitmask value). The structure
+    has the following format:
+
+    <\big-table|<tabular|<tformat|<cwith|6|8|1|-1|cell-tborder|1ln>|<cwith|6|8|1|-1|cell-bborder|1ln>|<cwith|6|8|1|-1|cell-lborder|0ln>|<cwith|6|8|1|-1|cell-rborder|0ln>|<cwith|5|5|1|-1|cell-bborder|1ln>|<cwith|1|3|1|-1|cell-tborder|1ln>|<cwith|1|3|1|-1|cell-bborder|1ln>|<cwith|1|3|1|-1|cell-lborder|0ln>|<cwith|1|3|1|-1|cell-rborder|0ln>|<cwith|4|4|1|-1|cell-tborder|1ln>|<cwith|9|9|1|-1|cell-tborder|1ln>|<cwith|8|8|1|-1|cell-bborder|1ln>|<cwith|10|10|1|-1|cell-bborder|1ln>|<cwith|9|10|1|1|cell-lborder|0ln>|<cwith|9|10|4|4|cell-rborder|0ln>|<cwith|1|1|1|-1|cell-tborder|0ln>|<cwith|1|1|1|1|cell-lborder|0ln>|<cwith|1|1|4|4|cell-rborder|0ln>|<cwith|2|2|1|-1|cell-tborder|1ln>|<cwith|1|1|1|-1|cell-bborder|1ln>|<cwith|2|2|1|-1|cell-bborder|1ln>|<cwith|3|3|1|-1|cell-tborder|1ln>|<cwith|2|2|1|1|cell-lborder|0ln>|<cwith|2|2|4|4|cell-rborder|0ln>|<table|<row|<cell|<strong|Type>>|<cell|<strong|Id>>|<cell|<strong|Description>>|<cell|<strong|Value>>>|<row|<cell|bytes>|<cell|1>|<cell|Block
+    header hash>|<cell|Sect. <reference|sect-blake2>>>|<row|<cell|bytes>|<cell|2>|<cell|Block
+    header (optional)>|<cell|Def. <reference|block>>>|<row|<cell|repeated>|<cell|3>|<cell|Block
+    body (optional)>|<cell|Def. <reference|sect-block-body>>>|<row|<cell|bytes>|<cell|>|<cell|>|<cell|>>|<row|<cell|bytes>|<cell|4>|<cell|Block
+    receipt (optional)>|<cell|>>|<row|<cell|bytes>|<cell|5>|<cell|Block
+    message queue (optional)>|<cell|>>|<row|<cell|bytes>|<cell|6>|<cell|Justification
+    (optional)>|<cell|Def. <reference|defn-grandpa-justification>>>|<row|<cell|bool>|<cell|7>|<cell|Indicates
+    whether the justification>|<cell|>>|<row|<cell|>|<cell|>|<cell|is empty
+    (i.e. should be ignored).>|<cell|>>>>>>
+      <strong|BlockData> Protobuf structure.
+    </big-table>
+  </definition>
+
+  <subsubsection|Transactions><label|sect-msg-transactions>
+
+  Transactions are sent directly in its full form to connected peers. It's
+  considered good behavior to implement a mechanism which only sends a
+  transaction once to each peer and avoids sending duplicates. Such a
+  mechanism is implementation specific and any absence of such a mechanism
+  can result in consequences which are undefined. The Polkadot Host much be
+  able to handle duplicate transactions received from its peers in order to
+  prevent the inclusion of duplicate transactions in multiple blocks.
```

Comment:
The transactions we receive jump to the transactions pool, which is also defined by these specs. There are quite a few steps involved before a transaction is included in a block.